aea/decision_maker/base.py

```diff
@@ -467,19 +486,14 @@ def _split_tx_fees(tx_fee: int) -> Dict[str, int]:
 class ProtectedQueue(Queue):
     """A wrapper of a queue to protect which object can read from it."""
 
-    def __init__(self, permitted_caller):
+    def __init__(self, access_code: str):
         """
         Initialize the protected queue.
 
-        :param permitted_caller: the permitted caller to the get method
+        :param access_code: the access code to read from the queue
         """
         super().__init__()
-        self._permitted_caller = permitted_caller
-
-    @property
-    def permitted_caller(self) -> "DecisionMaker":
-        """Get the permitted caller."""
-        return self._permitted_caller
+        self._access_code_hash = _hash(access_code)
```

Comment:
Note, we now use a hash here, the previous implementation meant the caller could be picked from the object and passed back in. Now one needs the access code which only the DM has. We could do something similar for the preferences and ownership states. Thoughts?

Reply:
OK although It is not true isolation, as it can be accessed anyway from the outside of the object. Also the `protected_get` can be circumvented.

Also `Inbox`/`Outbox`/`Multiplexer` have the same issue.

Reply:
I agree, not ideal.